RxJavaTemplate/app/src/test/java/co/nimblehq/rxjava/ui/screens/home/HomeViewModelTest.kt

```diff
@@ -20,45 +20,53 @@ class HomeViewModelTest {
     @Before
     fun setup() {
         When calling mockGetExampleDataUseCase.execute(any()) itReturns Single.just(MockUtil.data)
-        viewModel = HomeViewModel(
-            mockGetExampleDataUseCase
-        )
+        viewModel = HomeViewModel(mockGetExampleDataUseCase)
     }
 
     @Test
-    fun `When initializing, it should emit first data loading`() {
+    fun `When initializing HomeViewModel, it emits the first data correspondingly`() {
         val dataObserver = viewModel.data.test()
-        val loadingObserver = viewModel.showLoading.test()
 
         dataObserver
             .assertValueCount(1)
             .assertValue(MockUtil.data)
+    }
+
+    @Test
+    fun `When initializing HomeViewModel, it doesn't emit to show any loading indicator, defaulting to false`() {
+        val loadingObserver = viewModel.showLoading.test()
 
         loadingObserver
             .assertNoErrors()
-            .assertValues(false)
+            .assertValue(false)
```

Comment:
from my understanding here, we can't test the loading states correctly here because of `init` block executing. Should we apply the same way to use `TestScheduler` in DM project here to have a better test, the expected result should be `false, true, false`?

Reply:
and we need to update the test case name 😛

Reply:
hmm, I think this requirement is out of this ticket's scope man, maybe we can do that as a separate convention or improvement for sure 👍

Reply:
Okay, let's do it in a separate PR 🆗